kubernetes/linux/main.sh

```diff
@@ -259,16 +259,15 @@ else
 fi
 
 if [ $IS_GET_PODS_API_SUCCESS == 1 ]; then
-      podsResponse=$(cat podsResponseFile)
-      ITEMS_COUNT=$(echo $podsResponse | jq '.items | length')
+      ITEMS_COUNT=$(cat podsResponseFile | jq '.items | length')
       echo "found items count: $ITEMS_COUNT"
       if [ $ITEMS_COUNT -gt 0 ]; then
-            # exclude the pods which doesnt have containerId. could happen if the container fails to start because of bad image and tag etc..
-            podsWithValidContainerId=$(echo $podsResponse | jq -r '[.items[] | select( .status.containerStatuses != null and .status.containerStatuses[].containerID != null and  .status.containerStatuses[].containerID != "")]')
-            ITEMS_COUNT_WITH_CONTAINER_ID=$(echo $podsWithValidContainerId | jq '. | length')
-            if [ $ITEMS_COUNT_WITH_CONTAINER_ID -gt 0 ]; then
-                  containerRuntime=$(echo $podsWithValidContainerId | jq -r '.[0].status.containerStatuses[0].containerID' | cut -d ':' -f 1)
-                  nodeName=$(echo $podsWithValidContainerId | jq -r '.[0].spec.nodeName')
+            # select omsagent pods
```

Comment:
Do you the root cause and how will this fix the actual issue ?

Reply:
> Do you know the root cause and how will this fix the actual issue ?

> issue seems enumerating and validating large containerStatuses array. I have created issue on jq to confirm this issue. The fix is just checks for the our agent pods based on the metadata fields on any condition. I have validated on large JSON and empty JSON etc.. seems working fine. Planning to do more testing on this with larger size JSONs with many containers/pods etc..  I dont know customer has the same issue or not but we will have to confirm.  Seems having >100 containers per pod is not common.

Reply:
i am not sure about that, because our telemetry says there are few more clusters in this state. I suspect its because of something in the payload. So if you copy omsagent container 100 times in the payload does it repro ?

Reply:
> i am not sure about that, because our telemetry says there are few more clusters in this state. I suspect its because of something in the payload. So if you copy omsagent container 100 times (with different names) in the payload does it repro ?

> this is not issue with payload. on same payload fix in this pr or without this condition .status.containerStatuses[].containerID != null and  .status.containerStatuses[].containerID != "" works fine. So,  issue specific checking for containerID null or empty in ContainerStatuses array as the number of containers growing
.status.containerStatuses[].containerID != null and  .status.containerStatuses[].containerID != ""

since the expression is to enumerate containerStatuses array and check for containerID is not null and empty.
The result can be large if the containers and pods are too many.

updated fix is just to check our agent pods based on metadata and at max we will have two pods on node (rs and ds) scheduled.